### PR TITLE
Haitham dev

### PR DESCRIPTION
**Pull request by haithamAbuElnasr**

### CODE REVIEW

File: src/main/api/project.ts [hunk 1]
```diff
@@ -35,16 +35,17 @@ export function createProjectDir(name: string) {
...
-    archive: defaultToolObj(),
-    waybackurls_js: defaultToolObj(),
-    waybackurls_parameter: defaultToolObj(),
+    subFinder: defaultJobDetails,
+    liveDomains: defaultJobDetails,
+    screenWin: defaultJobDetails,
+    archive: defaultJobDetails,
+    waybackurls_js: defaultJobDetails,
+    waybackurls_parameter: defaultJobDetails,
   };
```
Comment (Mahmoudgalalz) [inline src/main/api/project.ts:49]: This is wrong structure

File: src/main/db/connect.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ export interface ProjectDetails {
     screens?: JobDetails;
     params?: JobDetails;
     liveDomains?: JobDetails;
+    archive?: JobDetails;
+    waybackurls_js?: JobDetails;
```
Comment (Mahmoudgalalz) [inline src/main/db/connect.ts:15]: we don't use snake case here 
wayBackUrlsJs

File: src/main/recon/httpx.ts [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
   try {
-    execSync(command);
+    const res = await execAsync(command);
+    const domainsFound = '0';
+    //  const domainsFound = resultFromStd(res.stderr, /\bFound (\d+) subdomains?/);
+    const db = connectJson(path.join(`${outputDir}/details.json`));
+    await db.update({
+      recon: {
+        screens: {
+          result: parseInt(domainsFound, 10),
```
Comment (Mahmoudgalalz) [inline src/main/recon/httpx.ts:56]: make it '' empty string

File: src/main/recon/waybackurls.ts [hunk 4]
```diff
@@ -17,34 +19,56 @@ export function wwayback(outputDir: string = PROJECT_DIR): {
   //  'type "D:\\recon_result\\domains.txt" | D:\\05-Haitham\\Projects\\hunter\\bin\\waybackurls >> "D:\\recon_result\\archive.txt"';
 
   try {
-    execSync(command);
+    const res = await execAsync(command);
+    const domainsFound = resultFromStd(res.stderr, /\bFound (\d+) subdomains?/);
+    const db = connectJson(path.join(`${outputDir}/details.json`));
+    await db.update({
```
Comment (Mahmoudgalalz) [inline src/main/recon/waybackurls.ts:25]: does this regex work ?!

File: src/main/recon/waybackurls.ts [hunk 5]
```diff
@@ -17,34 +19,56 @@ export function wwayback(outputDir: string = PROJECT_DIR): {
...
   const command = `cat "${path.join(
     outputDir,
     'recon_subdomins.txt',
   )}" | ${wwaybackPath} | findstr ".js" >> ${path.join(outputDir, 'waybackurls_js.txt')}`;
   try {
-    execSync(command);
+    const res = await execAsync(command);
+    const domainsFound = resultFromStd(res.stderr, /\bFound (\d+) subdomains?/);
+    const db = connectJson(path.join(`${outputDir}/details.json`));
+    await db.update({
```
Comment (Mahmoudgalalz) [inline src/main/recon/waybackurls.ts:53]: same as above

File: src/main/recon/waybackurls.ts [hunk 6]
```diff
@@ -55,7 +79,20 @@ export function parameter(outputDir: string = PROJECT_DIR): {
...
   )}`;
   try {
-    execSync(command);
+    const res = await execAsync(command);
+    const domainsFound = resultFromStd(res.stderr, /\bFound (\d+) subdomains?/);
+    const db = connectJson(path.join(`${outputDir}/details.json`));
+    await db.update({
+      recon: {
+        liveDomains: {
+          result: parseInt(domainsFound, 10),
```
Comment (Mahmoudgalalz) [inline src/main/recon/waybackurls.ts:88]: same as above

**Review by Mahmoudgalalz — CHANGES_REQUESTED** [review]
Its shit

**Comment by haithamAbuElnasr** [comment]
No tool outputs the counter but subfinder , the live domains outputs the domains themself i tryed to count them but already all method doesn't give the output in the run time ,onece i close the app the i find the output files i tryed to use the debug tool the app doesn't go after  the awit liveDomian() at all i tryed to put the subfinder command instated of the livedomaind command in same function of the liveDomain method it works normalygit add .

File: src/main/main.ts [hunk 7]
```diff
@@ -46,25 +46,25 @@ ipcMain.handle('httpx-live-domain', async (event, args) => {
...
-  return res;
+  event.returnValue = res;
 });
-ipcMain.handle('waybackurls-parameter', async (event, args) => {
+ipcMain.on('waybackurls-parameter', async (event, args) => {
   const { projectName } = args[0];
   const res = parameter(`${PROJECT_DIR}/${projectName}`);
-  return res;
+  event.returnValue = res;
 });
```
Comment (Mahmoudgalalz) [inline src/main/main.ts:68]: The handler are not correct

File: src/main/recon/httpx.ts [hunk 8]
```diff
@@ -17,12 +17,15 @@ export async function liveSubDomains(
...
+    const domainsFound = resultFromStdHTTPX(
+      res.stderr,
+      /^(https?:\/\/)?([a-zA-Z0-9.-]+)\.([a-zA-Z]{2,})(\/[^\s]*)?$/i,
+    );
     const db = connectJson(path.join(`${outputDir}/details.json`));
     await db.update({
       recon: {
         liveDomains: {
-          result: parseInt(domainsFound, 10),
+          result: domainsFound,
```
Comment (Mahmoudgalalz) [inline src/main/recon/httpx.ts:28]: This return an array, so it should be array.length()

**Review by Mahmoudgalalz — CHANGES_REQUESTED** [review]
the default values are not initiated, this is a good way, but need to work on the above requests

File: src/main/api/project.ts (hunk 1 above)
Comment (Mahmoudgalalz) [reply src/main/api/project.ts:49]: Still need to be inside recon objcet

**Comment by Mahmoudgalalz** [comment]
> No tool outputs the counter but subfinder , the live domains outputs the domains themself i tryed to count them but already all method doesn't give the output in the run time ,onece i close the app the i find the output files i tryed to use the debug tool the app doesn't go after the awit liveDomian() at all i tryed to put the subfinder command instated of the livedomaind command in same function of the liveDomain method it works normalygit add .

You need to pass the stderr not stdout as these tools do print in the error stream

File: src/main/recon/httpx.ts (hunk 8 above)
Comment (haithamAbuElnasr) [reply src/main/recon/httpx.ts:28]: I actually do it look to teh fun imp

**Comment by Mahmoudgalalz** [comment]
can you explain what issue you have !?

**Pull request closed without merging**